daemon/graphdriver/btrfs/btrfs.go args.lim.max_referenced undefined

### Description

While building docker-20.10.22 on Funtoo linux with btrfs-progs 6.1, got this error

> daemon/graphdriver/btrfs/btrfs.go:437:11: args.lim.max_referenced undefined (type _Ctype_struct_btrfs_qgroup_limit has no field or method max_referenced)

### Reproduce

1. extract source
2. hack/make.sh dynbinary

### Expected behavior

Build not fail.

### docker version

```bash
Not relevant.
```


### docker info

```bash
Not relevant.
```


### Additional Info

Only fails with btrfs-progs 6.1, with previous versions (6.0.x) builds ok.

IMHO the change of btrfs-progs that caused this affect is 

https://github.com/kdave/btrfs-progs/commit/03451430de7cd2fad18b0f01745545758bc975a5

--- a/daemon/graphdriver/btrfs/btrfs.go
+++ b/daemon/graphdriver/btrfs/btrfs.go
@@ -6,6 +6,9 @@ package btrfs // import "github.com/docker/docker/daemon/graphdriver/btrfs"
 /*
 #include <stdlib.h>
 #include <dirent.h>
+
+// keep struct field name compatible with btrfs-progs < 6.1.
+#define max_referenced max_rfer
 #include <btrfs/ioctl.h>
 #include <btrfs/ctree.h>
 
@@ -395,7 +398,7 @@ func subvolLimitQgroup(path string, size uint64) error {
 	defer closeDir(dir)
 
 	var args C.struct_btrfs_ioctl_qgroup_limit_args
-	args.lim.max_referenced = C.__u64(size)
+	args.lim.max_rfer = C.__u64(size)
 	args.lim.flags = C.BTRFS_QGROUP_LIMIT_MAX_RFER
 	_, _, errno := unix.Syscall(unix.SYS_IOCTL, getDirFd(dir), C.BTRFS_IOC_QGROUP_LIMIT,
 		uintptr(unsafe.Pointer(&args)))

